core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -209,23 +210,51 @@ public ServerBuilder port(int port, String protocol) {
 
     /**
      * Adds a new {@link ServerPort} that listens to the specified {@code port} of all available network
-     * interfaces using the specified {@link SessionProtocol}.
+     * interfaces using the specified {@link SessionProtocol}s. Specify multiple protocols to serve more than
+     * one protocol on the same port:
      *
-     * @deprecated Use {@link #http(int)} or {@link #https(int)}.
-     * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
+     * <pre>{@code
+     * ServerBuilder sb = new ServerBuilder();
+     * // Serve both HTTP and HTTPS at port 8080.
+     * sb.port(new InetSocketAddress(8080),
+     *         SessionProtocol.HTTP,
+     *         SessionProtocol.HTTPS);
+     * // Enable HTTPS with PROXY protocol support at port 8443.
+     * sb.port(new InetSocketAddress(8443),
+     *         SessionProtocol.PROXY,
+     *         SessionProtocol.HTTPS);
+     * }</pre>
      */
-    @Deprecated
-    public ServerBuilder port(int port, SessionProtocol protocol) {
-        ports.add(new ServerPort(port, protocol));
-        return this;
+    public ServerBuilder port(int port, SessionProtocol... protocols) {
+        return port(new ServerPort(port, protocols));
+    }
+
+    /**
+     * Adds a new {@link ServerPort} that listens to the specified {@code port} of all available network
+     * interfaces using the specified {@link SessionProtocol}s. Specify multiple protocols to serve more than
+     * one protocol on the same port:
+     *
+     * <pre>{@code
+     * ServerBuilder sb = new ServerBuilder();
+     * // Serve both HTTP and HTTPS at port 8080.
+     * sb.port(new InetSocketAddress(8080),
+     *         SessionProtocol.HTTP,
+     *         SessionProtocol.HTTPS);
```

Comment:
This method takes a `port` and `Iterable<SessionProtocol> protocols`, but the example is little different.